src/google-classroom-extractor/google_classroom_extractor/api/teachers.py

```diff
@@ -0,0 +1,147 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from typing import List, Dict, Optional
+import pandas as pd
+import sqlalchemy
+from googleapiclient.discovery import Resource
+from .api_caller import call_api
+
+
+def request_teachers(
+    resource: Optional[Resource], course_id: str
+) -> List[Dict[str, str]]:
+    """
+    Fetch Teachers API data for a course and return a list of teacher data
+
+    Parameters
+    ----------
+    resource: Optional[Resource]
+        a Google Classroom SDK Resource
+    course_ids: str
+        a Google Classroom course id as a string
+
+    Returns
+    -------
+    List[Dict[str, str]]
+        a list of Google Classroom Teacher resources,
+            see https://developers.google.com/classroom/reference/rest/v1/courses.teachers
+
+    DataFrame columns are:
+        courseId: Identifier of the course
+        userId: Identifier of the user
+        profile.id: Identifier of the user
+        profile.name.givenName: The user's first name
+        profile.name.familyName: The user's last name
+        profile.name.fullName: The user's full name formed by concatenating the first and last name values
+        profile.emailAddress: Email address of the user
+    """
+
+    assert isinstance(resource, Resource) or resource is None
+    assert isinstance(course_id, str)
+
+    if resource is None:
+        return []
+
+    return call_api(
+        resource.courses().teachers().list,  # type: ignore - courses() is dynamic
+        {"courseId": course_id},  # type: ignore - due to tail_recursive decorator
```

Comment:
mypy flags these ignores as invalid.

Reply:
fixed for mypy. pylance does complain, looking into.

Reply:
Hm, maybe there's something I'm still not doing "right" with Pylance, because it did not flag this.